File: package/main/src/Math/factorize.ts
/**
 * Prime factorization of a number
 * @param  {number} n Number to factorize
 * @returns number[] Array of prime factors
 * @example factorize(12); // [2, 2, 3]
 */
export const factorize = (n: number): number[] => {
  const result: number[] = [];
  let remaining = Math.abs(n);

  for (let factor = 2; factor * factor <= remaining; factor++) {
    while (remaining % factor === 0) {
      result.push(factor);
      remaining /= factor;
    }
  }

  if (remaining > 1) {
    result.push(remaining);
  }

  return result;
};

File: package/umt_python/src/math/factorize.py
def factorize(n: int) -> list[int]:
    """
    Prime factorization of a number.

    Args:
        n: Number to factorize.

    Returns:
        Array of prime factors.

    Example:
        >>> factorize(12)
        [2, 2, 3]
    """
    result: list[int] = []
    remaining = abs(n)

    factor = 2
    while factor * factor <= remaining:
        while remaining % factor == 0:
            result.append(factor)
            remaining //= factor
        factor += 1

    if remaining > 1:
        result.append(remaining)

    return result
